core/templates/dev/head/pages/exploration-editor-page/improvements-tab/feedback-improvement-task/feedback-improvement-task.directive.ts

```diff
@@ -32,43 +32,14 @@ angular.module('oppia').directive('feedbackImprovementTask', [
       controller: [
         '$scope', 'DateTimeFormatService', 'ThreadStatusDisplayService',
         function($scope, DateTimeFormatService, ThreadStatusDisplayService) {
-          $scope.getLatestMessage = function() {
-            var numMessages = $scope.getData().messages.length;
-            if (numMessages > 0) {
-              var latestMessage = $scope.getData().messages[numMessages - 1];
-              var threadIsNewlyOpened =
-                numMessages === 1 && latestMessage.updated_status === 'open';
-              return {
-                text: latestMessage.text,
-                author: latestMessage.author_username,
-                updatedOn: latestMessage.created_on,
-                updatedStatus: (
-                  threadIsNewlyOpened ? null : latestMessage.updated_status),
-              };
-            } else {
-              return {
-                text: '',
-                author: $scope.getData().originalAuthorName,
-                updatedOn: $scope.getData().lastUpdated,
-                updatedStatus: null,
-              };
-            }
+          $scope.getLastNonemptyMessageAuthorUsername = function() {
+            return $scope.getData().lastNonemptyMessageSummary.authorUsername ||
```

Comment:
Do you need the backup clause? 

If so, then maybe the name of this function is wrong (since presumably it happens in the case where the original author's message was empty)?

Reply:
Yes I do, not sure what a good name could be -- any suggestions?